packages/superset-ui-core/src/chart/registries/ChartBuildQueryRegistrySingleton.ts

```diff
@@ -1,7 +1,18 @@
-import { Registry, makeSingleton, OverwritePolicy, QueryContext } from '../..';
+import { Registry, makeSingleton, OverwritePolicy, QueryContext, SetDataMaskHook } from '../..';
 
 // Ideally this would be <T extends QueryFormData>
-type BuildQuery = (formData: any) => QueryContext;
+export type BuildQuery<T = any> = (
+  formData: T,
+  options?: {
+    extras?: {
+      cachedChanges?: any;
+    };
+    hooks?: {
+      setDataMask: SetDataMaskHook;
+      setCachedChanges: (newChanges: any) => void;
+    };
+  },
+) => QueryContext;
```

Comment:
I think it's possible to reuse `BuildQueryFunction` here:

```suggestion
type BuildQuery = BuildQueryFunction<QueryFormData & JsonObject>;
```

You just provide a more relaxed `FormData` type.